src/api/app/models/staging_project.rb

```diff
@@ -0,0 +1,142 @@
+class StagingProject < Project
+  has_many :staged_requests, class_name: 'BsRequest', foreign_key: :staging_project_id, dependent: :nullify
+  has_many :status_reports, through: :repositories
+  belongs_to :staging_workflow, inverse_of: :staging_projects
+
+  def staging_identifier
+    name[/.*:Staging:(.*)/, 1]
+  end
+
+  def untracked_requests
+    requests_to_review - staged_requests
+  end
+
+  # The difference between staged requests and requests to review is that staged requests are assigned to the staging project.
+  # The requests to review are requests which are not related to the staging project (unless they are also staged).
+  # They simply need a review from the maintainers of the staging project.
+  def requests_to_review
+    @requests_to_review ||= BsRequest.with_open_reviews_for(by_project: name)
+  end
+
+  def building_repositories
+    set_buildinfo unless @building_repositories
+    @building_repositories
+  end
+
+  def broken_packages
+    set_buildinfo unless @broken_packages
+    @broken_packages
+  end
+
+  def missing_reviews
+    return @missing_reviews if @missing_reviews
+
+    @missing_reviews = []
+    attribs = [:by_group, :by_user, :by_project, :by_package]
+
+    (requests_to_review + staged_requests).uniq.each do |request|
+      request.reviews.each do |review|
+        next if review.state.to_s == 'accepted' || review.by_project == name
+        # FIXME: this loop (and the inner if) would not be needed
+        # if every review only has one valid by_xxx.
+        # I'm keeping it to mimic the python implementation.
+        # Instead, we could have something like
+        # who = review.by_group || review.by_user || review.by_project || review.by_package
+        attribs.each do |att|
+          who = review.send(att)
+          if who
+            @missing_reviews << { id: review.id, request: request.number, state: review.state.to_s, package: request.first_target_package, by: who }
+          end
+        end
+      end
+    end
+
+    @missing_reviews
+  end
+
+  def overall_state
+    @overall_state ||= state
+  end
+
+  private
+
+  def state
+    return :empty unless staged_requests.exists?
+    return :unacceptable if untracked_requests.present? || staged_requests.obsolete.present?
+
+    overall_state = build_state
+    overall_state = check_state if overall_state == :acceptable
+
+    return :review if overall_state == :acceptable && missing_reviews.present?
+
+    overall_state
+  end
+
+  def build_state
+    set_buildinfo
+
+    return :building if building_repositories.present?
+    return :failed if broken_packages.present?
+
+    :acceptable
+  end
+
+  def relevant_status_reports
+    @relevant_status_reports ||= status_reports.where(uuid: repositories.map(&:build_id))
+  end
+
+  def missing_checks?
+    relevant_status_reports.any? { |report| report.missing_checks.present? }
+  end
+
+  def check_state
+    status_checks = Status::Check.where(status_reports_id: relevant_status_reports)
```

Comment:
`status_reports` instead `status_reports_id`

Reply:
You would get a `Mysql2::Error: Unknown column 'status_checks.status_reports'` then

Reply:
It is wrong defined here: https://github.com/openSUSE/open-build-service/pull/6224/files#diff-8d266b19374f8906dc972ad3d776f346R4

Reply:
Might be, but I don't find a fix for this right now... Having a look later.